### ISSUE TITLE
Prepare gRPC instrumentation donation for release

### ISSUE THREAD

**Issue by kaylareopelle** [open-telemetry/opentelemetry-ruby-contrib#1261]
Overall, the goal is to update the donated gRPC gem to match the other contrib projects' structure and tooling. This will allow us to use our existing CI that runs tests and releases gems with this new instrumentation.

TODOs:
- [x] Create a gRPC-specific example to replace the Gruf example
- [x] Move from Standard.rb to Rubocop and Rubocop-Performance to match our other gems
- [x] Migrate the tests from RSpec to Minitest
- [X] Add the gRPC gem to the relevant CI workflows (ci-instrumentation.yml) 
- [x] Add the gRPC gem to dependabot.yml
- [ ] Add the gRPC gem to the `opentelemetry-instrumentation-all` gem
- [x] Update the README to match our style
- [x] Update the Changelog for compatibility with toys
- [x] Update code of conduct to match OTel docs 
- [x] Remove RBS
- [x] Remove conflicting [Code of Conduct file](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/92d59eb33cf7fd58242cde22a7346960078ff29f/instrumentation/grpc/CODE_OF_CONDUCT.md)

Relates to the code merged in #1164

**Comment by michal-kazmierczak** [comment]
I believe we should now have completed the following tasks:
- Move from Standard.rb to Rubocop and Rubocop-Performance to match our other gems (https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1341)
- Migrate the tests from RSpec to Minitest (https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1330)
- Remove RBS (https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1341)
- Remove conflicting Code of Conduct file (https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1341)

By the way, I wanted to ask about `Update code of conduct to match OTel docs` - do instrumentation gems have a separate code of conduct? From what I can see, there is one global file in the root directory. I'm not sure if anything else than removing the conflicting file needs to be done.

**Comment by kaylareopelle** [comment]
Hi @michal-kazmierczak! Thanks for your gRPC example PR! I updated the issue to reflect the remaining TODOs. 

> By the way, I wanted to ask about Update code of conduct to match OTel docs - do instrumentation gems have a separate code of conduct? From what I can see, there is one global file in the root directory. I'm not sure if anything else than removing the conflicting file needs to be done.

You're right! Removing the conflicting doc is all that was needed.

**Comment by michal-kazmierczak** [comment]
Thank you for the update @kaylareopelle.

I've now posted another [PR](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1399), I believe the last one before the release. It addresses the following points:
- Add the gRPC gem to dependabot.yml
- Update the README to match our style
- Update the Changelog for compatibility with toys

Once it gets merged, I'll prepare a PR with the release.

**Comment by kaylareopelle** [comment]
That sounds great, @michal-kazmierczak! 

Forgive me if we've talked about this already, but would you be interested in becoming a code owner for the gRPC instrumentation going forward? 

By becoming a code owner, you would be added as a reviewer to any PR opened to update the `instrumentation/grpc` code. We may also tag you directly in issues related to gRPC instrumentation.

The OpenTelemetry Ruby team needs help from end users and other experts in the community to maintain our instrumentation. Your work on the gRPC instrumentation is invaluable, and we'd love to continue working with you.

**Comment by michal-kazmierczak** [comment]
> Forgive me if we've talked about this already, but would you be interested in becoming a code owner for the gRPC instrumentation going forward?

Absolutely!  I use gRPC in Ruby in my day-to-day work, and I’d be happy to help maintain and improve the instrumentation.

Please feel free to add me as a code owner - I’d be glad to review PRs and assist with any gRPC-related issues.

Looking forward to continuing to collaborate with the team. :)

**Comment by kaylareopelle** [comment]
Hi @michal-kazmierczak, @hibachrach  - Thanks for all your hard work to get this out the door! `opentelemetry-instrumentation-grpc` version 0.2.0 has been released 🎉
